Library/Homebrew/extend/pathname.rb

```diff
@@ -216,10 +216,10 @@ def cp_path_sub(pattern, replacement)
 
   # extended to support common double extensions
   def extname(path = to_s)
-    BOTTLE_EXTNAME_RX.match(path)
-    return $1 if $1
-    /(\.(tar|cpio|pax)\.(gz|bz2|lz|xz|Z))$/.match(path)
-    return $1 if $1
+    path =~ BOTTLE_EXTNAME_RX
+    return Regexp.last_match(1) if Regexp.last_match(1)
+    %r{(\.(dmg|tar|cpio|pax)\.(gz|bz2|lz|xz|Z|zip))$} =~ path
+    return Regexp.last_match(1) if Regexp.last_match(1)
```

Comment:
I think both the old and the new code could be improved and made more understandable by dropping reliance on magic global state and doing something like:

```ruby
bottle_ext = path[BOTTLE_EXTNAME_RX, 1]
return bottle_ext if bottle_ext
archive_ext = path[/(\.(tar|cpio|pax)\.(gz|bz2|lz|xz|Z))$/, 1]
return archive_ext if archive_ext
```